dbt copilot ide ga (#7011)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/dbt-copilot-faqs.md

sidebar_label: "dbt Copilot FAQs" 
description: "Answers to common questions about dbt Copilot."
---

# dbt Copilot FAQs <Lifecycle status="enterprise" /> 

<IntroText>
Read about common questions about dbt Copilot to understand how it works and how it can help you.
</IntroText>

dbt Copilot is a powerful AI engine that handles the tedious tasks, speeds up workflows, and ensures consistency, helping you deliver exceptional data products faster.

dbt Labs is committed to protecting your privacy and data. This page provides information about how the dbt Copilot AI engine handles your data. For more information, check out the [dbt Labs AI development principles](https://www.getdbt.com/legal/ai-principles) page.

## Overview 

<Expandable alt_header="What is dbt Copilot?">
dbt Copilot is a powerful artificial intelligence (AI) engine that's fully integrated into your dbt Cloud experience and designed to accelerate your analytics workflows. dbt Copilot embeds AI-driven assistance across every stage of the analytics development life cycle (ADLC), empowering data practitioners to deliver data products faster, improve data quality, and enhance data accessibility. 

> With automatic code generation, let dbt Copilot [generate code](/docs/cloud/use-dbt-copilot) using natural language, and [generate documentation](/docs/build/documentation), [tests](/docs/build/data-tests), [metrics](/docs/build/metrics-overview), and [semantic models](/docs/build/semantic-models) for you with the click of a button in the [dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-copilot) and, now in beta, [Visual Editor](/docs/cloud/build-ve-copilot).

Reviewer: Hiya @mirnawong1 

It could just be me, but the last link for the VE takes me to the dbt CoPilot page. Would it be better to link out to the VE page (/docs/cloud/visual-editor)?

Kind Regards
Natalie

Author: ah yea - do you think? i guess i was talking about it in the context of copilot for the ve. but can link to the ve general doc too

> With automatic code generation, let dbt Copilot [generate code](/docs/cloud/use-dbt-copilot) using natural language, and [generate documentation](/docs/build/documentation), [tests](/docs/build/data-tests), [metrics](/docs/build/metrics-overview), and [semantic models](/docs/build/semantic-models) for you with the click of a button in the [dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-copilot) and, now in beta, [Visual Editor](/docs/cloud/build-ve-copilot).

Author: ```suggestion
With automatic code generation, let dbt Copilot [generate code](/docs/cloud/use-dbt-copilot) using natural language, and [generate documentation](/docs/build/documentation), [tests](/docs/build/data-tests), [metrics](/docs/build/metrics-overview), and [semantic models](/docs/build/semantic-models) for you with the click of a button in the [dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-copilot) and, now in beta, [Visual Editor](/docs/cloud/use-visual-editor).
```

</Expandable>

<Expandable alt_header="Where can I find dbt Copilot?">

> dbt Copilot can currently be found in the [dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-copilot). Future releases will bring dbt Copilot to the [Visual Editor](/docs/cloud/build-ve-copilot) and even more parts of the dbt Cloud application.

Reviewer: Same here: https://github.com/dbt-labs/docs.getdbt.com/pull/7011/files#r1995346779

> dbt Copilot can currently be found in the [dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-copilot). Future releases will bring dbt Copilot to the [Visual Editor](/docs/cloud/build-ve-copilot) and even more parts of the dbt Cloud application.

Author: ```suggestion
dbt Copilot can currently be found in the [dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-copilot). Future releases will bring dbt Copilot to the [Visual Editor](/docs/cloud/use-visual-editor) and even more parts of the dbt Cloud application.
```

> To use dbt Copilot, you must have a dbt Cloud [Enterprise account](https://www.getdbt.com/contact) and and administrative privileges to opt-in to the feature for your team.

Reviewer: ```suggestion
To use dbt Copilot, you must have a dbt Cloud [Enterprise account](https://www.getdbt.com/contact) and administrative privileges to opt-in to the feature for your team.
```

</Expandable>

<Expandable alt_header="What are the benefits of using dbt Copilot?">

Use dbt Copilot to:

- Generate code from scratch or edit existing code with natural language.
- Generate documentation, tests, metrics, and semantic models for your models.
- Accelerate your development workflow with AI-driven assistance.

with a click of a button and ensuring data privacy and security.

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/dbt-copilot-doc.gif" width="100%" title="Example of using dbt Copilot to generate documentation in the IDE" />

</Expandable>

## Availability 

<Expandable alt_header="Who has access to dbt Copilot?" >

When enabled by an admin, dbt Copilot is available on a dbt Cloud [Enterprise account](https://www.getdbt.com/contact) to all dbt Cloud [developer license users](/docs/cloud/manage-access/seats-and-users).

</Expandable>

<Expandable alt_header="Is dbt copilot available for all deployment types?">

> Yes, dbt Copilot is powered by ai-codegen-api, which is deployed everywhere including [multi-tenant and single-tenant deployments](/docs/cloud/about-cloud/access-regions-ip-addresses).

Author: @grabowskit - do we want to specifically mentioned ai-codegen-api?

</Expandable>

<Expandable alt_header="Do we support BYOK (bring your own key) at the project level?">

The dbt Copilot BYOK option is currently an account-only configuration. However, there may be a future where we make this configurable on a project-level.

</Expandable>

## Privacy and data

<Expandable alt_header="Does dbt Copilot store or use personal data?">

The user clicks the dbt Copilot button. Aside from authentication, it works without personal data, but the user controls what is input into dbt Copilot.

</Expandable>

<Expandable alt_header="Does dbt Copilot access my warehouse data?">

To provide the services, dbt Copilot utilizes the data entered by the user in addition to metadata, including column names, model SQL, the model's name, and model documentation. The row-level data from the warehouse is never used or sent by dbt Labs to a third-party provider without action by the user. The user is responsible for double-checking output for completeness and accuracy.

</Expandable>

<Expandable alt_header="Can dbt Copilot data be deleted upon client written request?">

> To the extent client identifies personal or sensitive information uploaded by or on behalf of client to dbt Labs systems by the user in error, such data can be deleted within 30 days of written request.

Reviewer: ```suggestion
If the client identifies personal or sensitive information that was uploaded to dbt Labs systems in error by the user, such data can be deleted within 30 days of a written request.
```

Author: hey @nataliefiann ! this is a great sugg but i'm going to keep it as is since this was provided by legal

</Expandable>

<Expandable alt_header="Does dbt Labs own the output generated by dbt Copilot?">

> No, dbt Labs will not dispute your ownership of any code or artifacts unique to your company that are generated when you use dbt Copilot. Your code will not be used to train AI models for the benefit of dbt Labs or other third parties, including other dbt Labs customers. 

Reviewer: ```suggestion
No, dbt Labs will not dispute your ownership of any code or artifacts unique to your company that's generated when you use dbt Copilot. Your code will not be used to train AI models for the benefit of dbt Labs or other third parties, including other dbt Labs customers. 
```

</Expandable>

## Considerations

<Expandable alt_header="What are the considerations for using dbt Copilot?">

dbt Copilot has the following considerations to keep in mind:

- dbt Copilot is not available in the dbt Cloud CLI.
- dbt Copilot is not available in the dbt Cloud API.

> Future releases are planned that may bring dbt Copilot to even more parts of the dbt Cloud application

Reviewer: ```suggestion
Future releases are planned that may bring dbt Copilot to even more parts of the dbt Cloud application.
```
